src/ripple/app/misc/TxQ.h

```diff
@@ -149,9 +158,19 @@ class TxQ
         amendment is not enabled, OR if the account has no transactions
         in the queue.
     */
-    boost::optional<std::map<TxSeq, AccountTxDetails>>
+    boost::optional<std::map<TxSeq, AccountTxDetails const>>
     getAccountTxs(AccountID const& account, ReadView const& view) const;
 
+    /** Returns information about all transactions currently
+        in the queue.
+
+        @returns Uninitialized @ref optional if the FeeEscalation
+        amendment is not enabled, OR if there are no transactions
+        in the queue.
+    */
+    boost::optional<std::vector<TxDetails>>
```

Comment:
Just a question: I'm not sure what returning a `boost::optional` buys us here. Would the semantics of this function really suffer if they were:

    @returns If the FeeEscalation amendment is enabled, returns all transactions
             currently in the queue. An empty vector otherwise.